Add a simple method to disable the install of recommended packages

Upstream-Status: Pending

Usage:
   smart config --set ignore-all-recommends=1

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: smart-1.4.1/smart/transaction.py
===================================================================
--- smart-1.4.1.orig/smart/transaction.py
+++ smart-1.4.1/smart/transaction.py
@@ -611,7 +611,9 @@ class Transaction(object):
             for prv in req.providedby:
                 for prvpkg in prv.packages:
                     if not reqrequired:
-                        if pkgconf.testFlag("ignore-recommends", prvpkg):
+                        if sysconf.get("ignore-all-recommends", 0) == 1:
+                            continue
+                        elif pkgconf.testFlag("ignore-recommends", prvpkg):
                             continue
                     if isinst(prvpkg):
                         found = True
